### PR TITLE
Delay handling inline notification replies until the user session is established

### PR DESCRIPTION
**Pull request by stefanceriu**

### CODE REVIEW

File: ElementX/Sources/Application/AppCoordinator.swift [hunk 1]
```diff
@@ -313,40 +314,27 @@ class AppCoordinator: AppCoordinatorProtocol, AuthenticationFlowCoordinatorDeleg
             if let userSession {
                 userSession.clientProxy.roomsToAwait.insert(roomID)
             } else {
-                storedRoomsToAwait.insert(roomID)
+                storedRoomsToAwait = [roomID]
```
Comment (pixlwave) [inline ElementX/Sources/Application/AppCoordinator.swift:317]: Why is this a collection of rooms, and how come it is ok to wipe out the remaining contents of it?

File: ElementX/Sources/Application/AppCoordinator.swift [hunk 2]
```diff
@@ -474,6 +462,24 @@ class AppCoordinator: AppCoordinatorProtocol, AuthenticationFlowCoordinatorDeleg
...
+        }
+        
+        if let storedAppRoute {
+            userSessionFlowCoordinator.handleAppRoute(storedAppRoute, animated: false)
+        }
+        
+        if let storedInlineReply {
+            await processInlineReply(roomID: storedInlineReply.roomID, replyText: storedInlineReply.message)
+        }
+    }
```
Comment (pixlwave) [inline ElementX/Sources/Application/AppCoordinator.swift:482]: Would it make sense to set these all back to nil afterwards?

**Review by pixlwave — APPROVED** [review]
Good spot 👍

File: ElementX/Sources/Application/AppCoordinator.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Application/AppCoordinator.swift:482]: Isn't that what `@Consumable` is supposed to do? 🤔

File: ElementX/Sources/Application/AppCoordinator.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Application/AppCoordinator.swift:317]: I imagine it's a collection just for flexibility but I don't see how we can run multiple `notificationTapped` before consuming it 🤔

File: ElementX/Sources/Application/AppCoordinator.swift (hunk 2 above)
Comment (pixlwave) [reply ElementX/Sources/Application/AppCoordinator.swift:482]: Oh yep, I even looked at that and thought "we should replace this with movable types". 🤦‍♂️

Too early, *puts the kettle on*

**Pull request merged**